modules/kms/main.tf

```
@@ -1,15 +1,80 @@
+locals {
+  account_id        = data.aws_caller_identity.current.account_id
+  account_principal = "arn:aws:iam::${local.account_id}:root"
```

Let's use the data source here for govcloud

```suggestion
  account_principal = "arn:${data.aws_partition.current.partition}:iam::${local.account_id}:root"
```

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/partition